pandas/tseries/offsets.py

```diff
@@ -2123,6 +2123,54 @@ def onOffset(self, dt):
             return False
         return date(dt.year, dt.month, dt.day) == easter(dt.year)
 
+
+class CalendarDay(SingleConstructorOffset):
+    """
+    Calendar day offset. Respects calendar arithmetic as opposed to Day which
+    respects absolute time.
+    """
+    _adjust_dst = True
+    _inc = timedelta(days=1)
+    _prefix = 'CD'
+    _attributes = frozenset(['n', 'normalize'])
+
+    def __init__(self, n=1, normalize=False):
+        BaseOffset.__init__(self, n, normalize)
+
+    @apply_wraps
+    def apply(self, other):
+        """
+        Apply scalar arithmetic with CalendarDay offset. Incoming datetime
+        objects can be tz-aware or naive.
+        """
```

Comment:
what about adding a CD and a D which crosses dst?

Reply:
A test with CD that crosses dst exists here:
https://github.com/pandas-dev/pandas/blob/44349bfd8732419fbabd832e2e73ac4ca77e6c23/pandas/tests/tseries/offsets/test_offsets.py#L3189

A test with D that crosses dst exists here:
https://github.com/pandas-dev/pandas/blob/44349bfd8732419fbabd832e2e73ac4ca77e6c23/pandas/tests/tseries/offsets/test_offsets.py#L3056